sklearn/decomposition/dict_learning.py

```diff
@@ -110,7 +110,7 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         clf = Lasso(alpha=alpha, fit_intercept=False, normalize=False,
                     precompute=gram, max_iter=max_iter, warm_start=True)
         clf.coef_ = init
-        clf.fit(dictionary.T, X.T, check_input=False)
+        clf.fit(check_array(dictionary.T, order='F'), X.T, check_input=False)
```

Comment:
This will check for nan and inf values at each iteration in the dict learning routine. I would rather add a `check_input=True`  kwarg to sparse encode itself as well (to be passed along to `clf.fit`) and make the online dict learning routine call sparse_encode with `check_input=False` (assuming it the online dict learning routine manages the dictionary with the right memory alignment to begin with)